.github/workflows/lint.yml

```diff
@@ -0,0 +1,43 @@
+---
+name: Linting
+
+on:
+  push:
+    paths-ignore:
+      - 'aider/website/**'
+      - README.md
+      - HISTORY.md
+    branches:
+      - main
+  pull_request:
+    paths-ignore:
+      - 'aider/website/**'
+      - README.md
+    branches:
+      - main
+
+jobs:
+  mypy:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Check out repository
+      uses: actions/checkout@v4
+
+    - name: Set up Python 3.12
+      uses: actions/setup-python@v5
+      with:
+        python-version: "3.12"
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install '.[browser,dev,playwright,mypy]'
+
```

Comment:
add flake8 and black checks as they are in the pre-commit

Reply:
Better to just add https://github.com/pre-commit/action

Reply:
I'd like to keep this PR minimal and just about adding minimal typing and checks for types. So other linting workflows would be better added in separate PRs.